data_safe_haven/external/api/credentials.py

```diff
@@ -58,6 +63,35 @@ def decode_token(cls, auth_token: str) -> dict[str, Any]:
     def get_credential(self) -> TokenCredential:
         """Get a credential provider from the child class."""
 
+    def confirm_credentials_interactive(
+        self,
+        target_name: str,
+        user_name: str,
+        user_id: str,
+        tenant_name: str,
+        tenant_id: str,
+    ) -> None:
+        """
+        Allow user to confirm that credentials are correct.
+
+        Responses are cached so the user will only be prompted once per run.
+        If 'skip_confirmation' is set, then no confirmation will be performed.
+
+        Raises:
+            DataSafeHavenValueError: if the user indicates that the credentials are wrong
+        """
+        if self.skip_confirmation:
+            return
+        if (user_id, tenant_id) in DeferredCredential.cache_:
+            return
+        DeferredCredential.cache_.add((user_id, tenant_id))
+        self.logger.info(f"You are logged into the [blue]{target_name}[/] as:")
+        self.logger.info(f"\tuser: [green]{user_name}[/] ({user_id})")
+        self.logger.info(f"\ttenant: [green]{tenant_name}[/] ({tenant_id})")
+        if not console.confirm("Are these details correct?", default_to_yes=True):
+            msg = "Selected credentials are incorrect."
+            raise DataSafeHavenValueError(msg)
```

Comment:
Let's make sure we test this method, there are quite a few paths it can take.

Reply:
Done in 8092290ea